config/environments/development.rb

```diff
@@ -67,12 +71,13 @@
   config.api.v1                                  = ActiveSupport::OrderedOptions.new
   config.api.v1.connection_options               = ActiveSupport::OrderedOptions.new
   config.api.v1.connection_options.namespace     = 'Limber'
-  config.api.v1.connection_options.url           = ENV.fetch('API_URL', "http://#{localhost}:3000/api/1/")
+  config.api.v1.connection_options.url           = rewrite_localhost(ENV.fetch('API_URL', 'http://localhost:3000/api/1/'))
   config.api.v1.connection_options.authorisation = ENV.fetch('API_KEY', 'development')
 
   config.api.v2                                  = ActiveSupport::OrderedOptions.new
   config.api.v2.connection_options               = ActiveSupport::OrderedOptions.new
-  config.api.v2.connection_options.url           = ENV.fetch('API2_URL', "http://#{localhost}:3000/api/v2")
+  config.api.v2.connection_options.url           = rewrite_localhost(ENV.fetch('API2_URL', 'http://localhost:3000/api/v2'))
```

Comment:
I think we are trying to reuse API2_URL but would be better to have API_2_JS as another env variable.
If I wanted to change the settings in my local to do something different from the default I would try to redefine all environment variables needed with the values needed, but this rewrite could modify the user settings.
(I'm talking only about the rewrite in line 79)
```suggestion
  config.api.v2.connection_options.url           = ENV.fetch('API2_JS_URL', rewrite_localhost('http://localhost:3000/api/v2'))
```

Reply:
If you haven't also declared the environment variable `LOCALHOST` then no rewrite will occur.  Docker Compose sets this environment variable so that Docker will use the correct localhost for the correct places in the code.  I think adding another environment variable such as `API_2_JS` will only make explaining this behaviour even more complicated for developers.  The only reason for the rewrite is so that the endpoint, which should be the same for both server-side and client-side requests, has the correct URL associated with it in each location.  I can't think of a scenario when the developer might want to have server-side talk to one Sequencescape instance and client-side talk to another instance.  If you set `API2_URL` environment variable to:

- https://localhost:9999/api/v2 then it would be changed to https://host.docker.internal:9999/api/v2 for server side requests from inside the container and left as `https://localhost:9999/api/v2` for client side requests in the browser.
- https://sequencescape.com/api/v2 then it would not be modified at all for either server-side or client-side requests.

Reply:
If you haven't also declared the environment variable `LOCALHOST` then no rewrite will occur.  Docker Compose sets this environment variable so that Docker will use the correct localhost for the correct places in the code.  I think adding another environment variable such as `API_2_JS` will only make explaining this behaviour even more complicated for developers.  The only reason for the rewrite is so that the endpoint, which should be the same for both server-side and client-side requests, has the correct URL associated with it in each location.  I can't think of a scenario when the developer might want to have server-side talk to one Sequencescape instance and client-side talk to another instance.  If you set `API2_URL` environment variable to:

- https://localhost:9999/api/v2 then it would be changed to https://host.docker.internal:9999/api/v2 for server side requests from inside the container and left as `https://localhost:9999/api/v2` for client side requests in the browser.
- https://sequencescape.com/api/v2 then it would not be modified at all for either server-side or client-side requests.